### PR TITLE
configuring timeout for graceful close.

### LINKED ISSUE #780: Extreme slowdowns (x20) in web spec tests with warp==3.3.4
### Bug Report

We are running some webspecs with a scotty/warp/wreq/http2 stack.

The freeze file can be seen at: https://gist.github.com/hasufell/0d709bbbdc06af9a86809d64be69dd6c#file-cabal-freeze-broken

Our entire test suite used to take **8 minutes**, with the bump to `warp==3.3.4` it takes about **3 hours**. There is no other bump that can cause it, *all other dependencies are exactly the same*.

* OS name and version: ubuntu bionic docker containers (reproducible outside of docker as well)
* Tools: ghc-8.6.5, cabal-3

I wasn't able to create a minimal test case yet, please bear with me.

### PR DESCRIPTION
**Pull request by kazu-yamamoto**
This should fix #780.

@hasufell Would you test this code? I did not test this code so much.

@snoyberg This code changes the convention of `Connection`. Is this acceptable?

### CODE REVIEW

File: warp/Network/Wai/Handler/Warp.hs [hunk 1]
```diff
@@ -72,13 +72,17 @@ module Network.Wai.Handler.Warp (
   , setLogger
   , setServerPushLogger
   , setGracefulShutdownTimeout
+  , setGracefulCloseTiemout1
```
Comment (avkapustin) [inline warp/Network/Wai/Handler/Warp.hs]: Sorry, probably mistyping? setGracefulCloseTi**me**out1 ?

Comment (kazu-yamamoto) [reply warp/Network/Wai/Handler/Warp.hs]: Nice catch! Fixed.

**Comment by kazu-yamamoto** [comment]
@snoyberg Thank you for reviewing.
@hasufell If you are fine with this fix, I will merge this.

**Comment by hasufell** [comment]
@kazu-yamamoto I tested it, this fixed it.

**Comment by kazu-yamamoto** [comment]
@hasufell Thank you. Merged.

**Comment by fosskers** [comment]
Are you planning to make a release with this included?

**Comment by kazu-yamamoto** [comment]
I noticed that we should update warp-tls, too. Just a moment.

**Comment by kazu-yamamoto** [comment]
I have released new warp and warp-tls.

**Comment by fosskers** [comment]
Thank you! I'll upgrade to these right away.

**Pull request merged**